pydoctor/test/test_sphinx.py

```diff
@@ -66,12 +66,20 @@ def inv_reader_nolog() -> sphinx.SphinxInventory:
 
 @pytest.fixture
 def inv_writer() -> InvWriter:
-    return InvWriter(logger=PydoctorLogger(), project_name='project_name')
+    return InvWriter(
```

Comment:
I don't like per-test fixtures as I think that they make the code harder to read.

I like the module or package fixtures.

I would argue that they can be replaced by a simple function that is explicitly called inside each test.
In this way, you can pass extra arguments to affect the execution of the test... and you also have the function call for inspection with your IDE (if you use any) 

With the current approach, you read a test and see the assertion of 'project-name' or '1.2.0.rc1` and you will ask: from where did those  value came?

If the test don't care about the value... is ok to use the default.

```python
def get_writer(name=None, version=None, logger=None):
  if name is None:
    name = 'project_name'
  if version is None:
    version = '1.2.0.rc1'
  if  logger is None:
    logger = PydoctorLogger()
  return InvWriter(logger=logger, project_name=name, project_version=version)

# When we have multiple calls with similar arguments.
def get_writer_without_log(name=None, version=None):
  return get_writer(name=name, version=version, logger=PydoctorNoLogger())

def test_error_on_writing():
   writer = get_writer()
   # Trigger an error in which name and version don't matter.


def test_project_name_and_version_in_inventory():
    writer = get_writer(name='acme-lib', version='1.3.0')
    # have assertion based on values defined in the same tests.

```

Reply:
> I would argue that they can be replaced by a simple function that is explicitly called inside each test.

I tend to prefer that style

Reply:
Please check the current implementation... it's no longer a fixture, but is implemented a bit different so that you don't have to touch the private logger attribute.